src/libexpr/primops.cc

```diff
@@ -4154,7 +4154,14 @@ void EvalState::createBaseEnv()
                     .arity = std::max(primOp.args.size(), primOp.arity),
                     .name = primOp.name,
                     .args = primOp.args,
-                    .doc = primOp.doc,
+                    .doc =
+                        primOp.experimentalFeature
+                        ? strdup(
```

Comment:
Maybe we should just change the type of `doc` to `std::string` or `BackedStringView`, then we don't need hacks like this.

Reply:
In 9800c1e8074d248f75ea9bed1b5a0f76e799863d I went with the appraoch of just dumping the info in the JSON, and letting the remote side render it.

I think that is generally preferable, but in this case there is also a repl way of viewing the docs, so we would have to do something in two places.

Maybe that's OK --- e.g. I don't know if the repl one supports markdown necessarily, but didn't want to not "come clean" on that caveat :)

Reply:
My https://github.com/NixOS/nix/pull/8330 now does what I suggested there too.